rasa/core/train.py

```diff
@@ -173,17 +173,14 @@
 
 
 def do_interactive_learning(
-    args: argparse.Namespace,
-    stories: Optional[Text] = None,
-    additional_arguments: Dict[Text, typing.Any] = None,
+    args: argparse.Namespace, file_importer: Optional[TrainingDataImporter] = None,
```

Comment:
why not this? If you keep it `Optional` you have to handle that down in `record_messages`
```suggestion
    args: argparse.Namespace, file_importer:TrainingDataImporter,
```

Reply:
wasn't sure why the stories were optional there, I kept it consistent to how it was, didn't know if that was intentional because of another functionality using the lower methods. Will take a deeper look into that

Reply:
Looks like it's truly optional all the way down to `load_data` where it has `if resource_name:`, but honestly don't know quite when/if setting the default `= None` makes sense for an optional param?

Reply:
No, I checked the usages and it's just used in the one place where have a proper object and not none

Reply:
i'm confused 😄as far as i can see, in `load_data` it should be optional bc that method handles both cases. and then all the methods further out would also be optional?

Reply:
just change the ones you touched (because it's not optional there) and leave the rest (potentially create an issue)

Reply:
got it, i see what you mean now, you have to provide a file importer to create the visualization, but not necessarily for the methods underneath. duh